Assets/_BForBoss/_Weapons/Scripts/Weapons/Weapon.cs

```diff
@@ -25,22 +26,23 @@ public bool ActivateWeapon
                  OnSetWeaponActivate?.Invoke(value);
              }
          }
-         
+
          public event Action<int> OnFireWeapon;
          public event Action<bool> OnSetWeaponActivate;
-         
+
          public int AmmunitionAmount => _ammunitionAmount;
          public int MaxAmmunitionAmount => _weaponProperties.AmmunitionAmount;
          public float MaxReloadDuration => _weaponProperties.ReloadDuration;
          public float ElapsedReloadDuration => _elapsedReloadDuration;
          public string NameOfWeapon => _weaponProperties.NameOfWeapon;
          public bool IsRayCastingWeapon => _weaponProperties.IsRayCastingWeapon;
+         public float DamagePerRayCast => IsRayCastingWeapon ? _weaponProperties.DamagePerRayCast : 0;
```

Comment:
Nitpick: is `PerRayCast` necessary in the variable name? Makes it sound like there is a Damage not by raycast  that can be associated with the weapon?

it might just be better to rename to  `Damage`. What do you think?

Reply:
Yeah, since if it's not a raycast weapon, then the damage will appear in projectile bullet.